src/Microsoft.Health.Fhir.Core/Features/Operations/Export/CreateExportRequestHandler.cs

```diff
@@ -77,6 +77,29 @@ public async Task<CreateExportResponse> Handle(CreateExportRequest request, Canc
             // Otherwise, we will create a new export job. This will be a best effort since the likelihood of this happen should be small.
             ExportJobOutcome outcome = await _fhirOperationDataStore.GetExportJobByHashAsync(hash, cancellationToken);
 
+            var filters = new List<ExportJobFilter>();
+
+            if (!string.IsNullOrWhiteSpace(request.Filters))
+            {
+                var filterArray = request.Filters.Split(",");
+                foreach (string filter in filterArray)
+                {
+                    var parameterIndex = filter.IndexOf("?", StringComparison.Ordinal);
+                    var filterType = filter.Substring(0, parameterIndex);
+
+                    var filterParameters = filter.Substring(parameterIndex + 1).Split("&");
+                    var parameterTupleList = new List<Tuple<string, string>>();
+
+                    foreach (string parameter in filterParameters)
+                    {
+                        var keyValue = parameter.Split("=");
+                        parameterTupleList.Add(new Tuple<string, string>(keyValue[0], keyValue[1]));
+                    }
+
+                    filters.Add(new ExportJobFilter(filterType, parameterTupleList));
+                }
```

Comment:
This looks quite complex, can it be make a separate class/method which can be unit tested?

Reply:
I can move this to a private method, but I don't think it should be public since it is only used here. There are some new unit tests for this code in CreateExportRequestHandlerTests.cs